### PR TITLE
Implement Value as separate entity in Python FOCS parser

### PR DESCRIPTION
**Pull request by o01eg**
Known issue:

```
                SetStockpile value = Value + min(abs(Value(Target.MaxStockpile) - Value), 1.000000) * (1.000000 - 2.000000 * Value ? Value(Target.MaxStockpile))
```
became
```
                SetStockpile value = Value + min(abs(Value(Target.MaxStockpile) - (Value) // StaticCast{i,d}
                    ), 1.000000) * (1.000000 - 2.000000 * Value(Target.MaxStockpile) ? Value)
```

### CODE REVIEW

File: parse/PythonParser.cpp [hunk 1]
```diff
@@ -153,6 +155,15 @@ PythonParser::PythonParser(PythonCommon& _python, const boost::filesystem::path&
...
         py::class_<value_ref_wrapper<Visibility>>("ValueRefVisibility", py::no_init);
+        py::class_<value_placeholder>("ValuePlaceHolder", py::no_init)
+            .def("__call__", &value_placeholder::call<double>)
+            .def(py::self_ns::self + double())
+            .def(py::self_ns::self * double())
+            .def(py::self_ns::self / double())
+            .def(py::self_ns::self - int())
+            .def(int() - py::self_ns::self)
+            .def(py::self_ns::self * int())
+            .def(py::self_ns::self / int());
```
Comment (Cjkjvfnby) [inline parse/PythonParser.cpp:166]: Should we just implement all 16 operations?

`(+ - / *) X (int, float) X {direct, reverse}`

A bit of boilerplate but will save time in the future. 

If it's possible to put this into a separate file, I can write a script that will generate it.  Or maybe C++ has some metaprogramming features to do it.

Comment (o01eg) [reply parse/PythonParser.cpp:166]: Yes, we should but I have no idea how to declare it.

**Comment by o01eg** [comment]
I got no idea how to enforce auto-typing to use `Variable<double>` instead of `Variable<int>` in 
```python
Value(Target.MaxStockpile) - Value
```

maybe I should just use separate names for different types of `Value` variable.

**Comment by Vezzra** [comment]
@o01eg, closed as invalid/superseded/abandoned/...?

**Comment by o01eg** [comment]
Closed as invalid.

**Pull request closed without merging**